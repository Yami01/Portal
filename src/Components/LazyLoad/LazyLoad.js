/**
 *
 * LazyLoad
 * @format
 *
 */

import type {AbstractComponent, Node} from 'react'
import React, {memo} from 'react'
import {View} from 'react-native'
import SkeletonPlaceholder from 'react-native-skeleton-placeholder'

import type {PropsType} from './types'
import {Colors} from '@/Theme/Variables'

const LazyLoad: AbstractComponent<PropsType> = memo(
  (props: PropsType): Node => {
    const { style, children, backgroundColor, highlightColor, speed, testID } =
      props

    return (
      <View style={style} testID={testID}>
        <SkeletonPlaceholder
          backgroundColor={backgroundColor || Colors.gray}
          highlightColor={highlightColor || Colors.gray}
          speed={speed || 800}
        >
          {children}
        </SkeletonPlaceholder>
      </View>
    )
  },
)

export default LazyLoad
