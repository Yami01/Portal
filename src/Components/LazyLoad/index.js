/**
 *
 * @flow
 *
 */

export {default} from './LazyLoad';
