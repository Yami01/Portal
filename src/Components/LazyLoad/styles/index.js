/**
 *
 * @format
 *
 */

import {StyleSheet} from 'react-native'

const LazyLoadStyles: { ... } = StyleSheet.create({})

export default LazyLoadStyles
