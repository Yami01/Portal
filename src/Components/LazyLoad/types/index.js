/**
 *
 * @format
 *
 */

import type {ViewStyleProp} from 'react-native/Libraries/StyleSheet/StyleSheet'
import type {Node} from 'react'

export type PropsType = {
  style?: ViewStyleProp,
  children?: Node,
  backgroundColor?: string,
  highlightColor?: string,
  speed?: number,
  testID?: string,
}
