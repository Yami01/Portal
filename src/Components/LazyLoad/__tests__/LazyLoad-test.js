/**
 *
 * LazyLoad Test
 * @format
 * @flow
 *
 */

import React from 'react'
import renderer from 'react-test-renderer'

import LazyLoad from '../LazyLoad'

test('LazyLoad', () => {
  const tree = renderer.create(<LazyLoad />).toJSON()
  expect(tree).toMatchSnapshot()
})
