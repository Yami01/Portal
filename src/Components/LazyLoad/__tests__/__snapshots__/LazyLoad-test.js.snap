// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyLoad 1`] = `
<View>
  <View />
</View>
`;
